src/file.rs

```diff
@@ -93,6 +98,24 @@ impl File {
     }
 }
 
+/// Read the content and return as a [String] if it can be decoded as one of the supported encodings from [Encoding].
+pub fn read(input: impl Read) -> Result<String, FileError> {
+    read_to_text_data(input).map(|content| content.data)
+}
+
+/// Read the contents of a file from the given path and return as a [String] if it can be decoded as one of the supported encodings from [Encoding].
+pub fn read_to_string(path: impl AsRef<Path>) -> Result<String, FileError> {
```

Comment:
The name of this feels slightly awkward because `read` is **effectively** a `read_to_string`. Personally, I'd change `read` to `read_to_string` and call this something like `read_from_path`.

Reply:
Hmm... after more thought, I get the comparison between this and `std::fs::read_to_string`. It's the `read` not matching `std::fs::read` that feels awkward. That's not a huge deal. There's no requirement for this library to follow `std::fs`.

Reply:
Maybe my original thought of `read_from_reader` then?